src/pages/SupplierRequisitionsPage.js

```diff
@@ -38,36 +40,44 @@ export class SupplierRequisitionsPage extends React.Component {
     };
   }
 
-  onDeleteConfirm = () => {
+  componentDidMount() {
+    const { settings, database } = this.props;
+    this.setState({ usesPrograms: !!getAllPrograms(settings, database) });
+  }
+
+  onDelete = shouldConfirm => () => {
     const { selection } = this.state;
     const { database } = this.props;
-    database.write(() => {
-      const requisitionsToDelete = [];
-      for (let i = 0; i < selection.length; i += 1) {
-        const requisition = this.requisitions.find(
-          currentRequisition => currentRequisition.id === selection[i]
-        );
-        if (requisition.isValid() && !requisition.isFinalised) {
-          requisitionsToDelete.push(requisition);
+
+    if (shouldConfirm) {
+      database.write(() => {
+        const requisitionsToDelete = [];
+        for (let i = 0; i < selection.length; i += 1) {
+          const requisition = this.requisitions.find(
+            currentRequisition => currentRequisition.id === selection[i]
+          );
+          if (requisition.isValid() && !requisition.isFinalised) {
+            requisitionsToDelete.push(requisition);
+          }
         }
-      }
-      database.delete('Requisition', requisitionsToDelete);
-    });
-    this.setState({ selection: [] });
-    this.refreshData();
-  };
+        database.delete('Requisition', requisitionsToDelete);
+      });
+    }
 
-  onDeleteCancel = () => {
     this.setState({ selection: [] });
     this.refreshData();
   };
 
-  onNewRequisition = otherStoreName => {
+  onNewRequisition = requisitionValues => {
     const { database, currentUser } = this.props;
-
     let requisition;
+    if (!requisitionValues) {
+      this.setState({ byProgramModalOpen: false, isCreatingRequisition: false });
+      return;
+    }
     database.write(() => {
-      requisition = createRecord(database, 'Requisition', currentUser, otherStoreName);
+      requisition = createRecord(database, 'Requisition', currentUser, requisitionValues);
+      if (requisition.program) requisition.addItemsFromProgram(database);
```

Comment:
do we close the modal if !!requisitionValues ?

Reply:
Gets closed in `navigateToRequisition` [ does cleanup, clears the selection etc. ]

Reply:
does it work with 'original' way of creating requisition (it might, but I don't see it immediately in code as state.isCreatingRequisition = false is not set as far as i can see, could be wrong)